### PR TITLE
Extend interface.rb: add ipv4_forwarding

### PR DESCRIPTION
**Pull request by chrisvanheuveln**
* simple boolean property
* Type, Prov., Beaker, Demo, README, CHANGELOG
* I moved some of the ACL code to keep the references in alpha order but I didn't change any of that code

### CODE REVIEW

**Comment by mikewiebe** [comment]
:+1:  Assuming the conflicts get resolved.

File: README.md [hunk 1]
```diff
@@ -1627,24 +1627,7 @@ Shutdown state of the interface. Valid values are 'true', 'false', and
...
-prior this configuration. Valid values are string, keyword 'default'.
-
-###### `ipv6_acl_in`
-Apply ipv6 access list on the interface in ingress direction. Access-list should be present on the network device
-prior this configuration. Valid values are string, keyword 'default'.
-
-###### `ipv6_acl_out`
-Apply ipv6 access list on the interface in egress direction. Access-list should be present on the network device
-prior this configuration. Valid values are string, keyword 'default'.
+Switchport mode of the interface. Interfaces that support `switchport_mode` may default to layer 2 or layer 3 depending on platform, interface type, or the `system default switchport` setting. An interface may be explicitly set to Layer 3 by setting `switchport_mode` to 'disabled'. Valid values are 'disabled', 'access', 'tunnel', 'fex_fabric', 'trunk', 'fabricpath' and 'default'. This property is not supported on IOS XR.
```
Comment (chrisvanheuveln) [inline README.md]: I cleaned up a couple of entries in the README while fixing the sync conflicts. 
I did not make any additional code changes with this sync.

File: tests/beaker_tests/cisco_interface/test_interface.rb [hunk 2]
```diff
@@ -173,6 +175,8 @@
     ipv4_netmask_length:           31,
     ipv4_address_secondary:        '2.2.2.2',
     ipv4_netmask_length_secondary: 31,
+    ipv4_forwarding:               true,
+    ipv4_pim_sparse_mode:          true,
```
Comment (glennmatthews) [inline tests/beaker_tests/cisco_interface/test_interface.rb]: These belong in L3_misc_nexus (below), not in L3_misc

File: README.md [hunk 3]
```diff
@@ -1705,6 +1691,24 @@ Network mask length of the secondary IP address on the interface. Valid values a
 ###### `ipv4_arp_timeout`
 Address Resolution Protocol (ARP) timeout value. Valid values are integer and keyword 'default'. Currently only supported on vlan interfaces. This property is not supported on IOS XR as IOS XR does not support vlan interfaces.
 
+###### `ipv4_forwarding`
+IP forwarding state.  Valid values are string or keyword 'default'.
```
Comment (glennmatthews) [inline README.md]: "This property is not supported on IOS XR". Also please update the caveat table for this provider to include this same limitation.

**Comment by glennmatthews** [comment]
Do you want to add this property in demo_interface?

**Comment by mikewiebe** [comment]
@chrisvanheuveln Agree with @glennmatthews comments.  I sent you an out of band email earlier today to add any needed XR considerations but should have added them here 2.

**Comment by chrisvanheuveln** [comment]
@glennmatthews It was in demo_interface in the first commit. I guess my sync removed it. I'll fix.

File: tests/beaker_tests/cisco_interface/test_interface.rb (hunk 2 above)
Comment (chrisvanheuveln) [inline tests/beaker_tests/cisco_interface/test_interface.rb]: More sync fallout. Fixed.

**Comment by chrisvanheuveln** [comment]
I see my NU PR hasn't been merged yet so let's hold off on this merge til that's been done. Thanks.

**Comment by glennmatthews** [comment]
:+1:

**Pull request merged**